parsers/json/json.cc

```diff
@@ -82,6 +86,664 @@ namespace
     }
     return lhs->location().view() == rhs->location().view();
   }
+
+  struct DebugJson
+  {
+    Node value;
+  };
+
+  static std::ostream& operator<<(std::ostream& os, const DebugJson& debug)
+  {
+    return os << json::to_string(debug.value);
+  }
+
+  namespace pointer
+  {
+    std::optional<std::string> unescape_pointer(const std::string_view& pointer)
+    {
+      std::ostringstream os;
+      size_t index = 0;
+      while (index < pointer.size())
+      {
+        char c = pointer[index];
+        if (c == '~')
+        {
+          if (index + 1 == pointer.size())
+          {
+            logging::Error() << "Invalid `~` in pointer";
+            return std::nullopt;
+          }
+
+          char i = pointer[index + 1];
+          if (i == '0')
+          {
+            os << '~';
+          }
+          else if (i == '1')
+          {
+            os << '/';
+          }
+          else
+          {
+            logging::Error() << "Invalid escape value '" << i << "'";
+            return std::nullopt;
+          }
+
+          index += 2;
+          continue;
+        }
+
```

Comment:
Perhaps, check that this hasn't been called with a `string_view` containing `/`.
```patch

assert(c != '/');
```